frame/contracts/src/wasm/runtime.rs

```diff
@@ -1043,10 +1083,51 @@ define_env!(Env, <E: Ext>,
 		output_len_ptr: u32
 	) -> ReturnCode => {
 		ctx.call(
-			CallFlags::from_bits(flags).ok_or_else(|| "used reserved bit in CallFlags")?,
-			callee_ptr,
-			gas,
-			value_ptr,
+			CallFlags::from_bits(flags).ok_or_else(|| Error::<E::T>::InvalidCallFlags)?,
+			CallType::Call{callee_ptr, value_ptr, gas},
+			input_data_ptr,
+			input_data_len,
+			output_ptr,
+			output_len_ptr,
+		)
+	},
+
+	// Execute code in the context of the current contract.
+	//
+	// The code is executed in the same storage transaction as the caller.
```

Comment:
Sorry I had a brain fart when suggesting this line. I works the same as for the usual call: The transaction is spawned when the pushed frame is executed.

```suggestion
```

Reply:
My fault, I should've noticed that.